internal/authtoken/options.go

```diff
@@ -91,3 +95,21 @@ func WithPublicId(id string) Option {
 		o.withPublicId = id
 	}
 }
+
+// WithPasswordOptions allows passing through options for the password package.
+// This is useful for things like tests where you may use testing helper
+// functions in this package but they also create password resources.
+func WithPasswordOptions(with ...password.Option) Option {
+	return func(o *options) {
+		o.withPasswordOptions = with
+	}
+}
+
+// WithIamOptions allows passing through options for the IAM package. This is
+// useful for things like tests where you may use testing helper functions in
+// this package but they also create IAM resources.
+func WithIamOptions(with ...iam.Option) Option {
+	return func(o *options) {
+		o.withIamOptions = with
+	}
+}
```

Comment:
Should these go in `testing.go`?

Reply:
The reason I didn't -- and maybe this is wrong -- is that they do go into normal non-testing functions, even if they're mostly used for tests. So I didn't want to have e.g. `NewAccount(..., ..., ..., password.TestOptions...)`. If these were only ever used in actual testing functions I would definitely put them in testing.go and label them e.g. `WithTestIamOption` and `WithTestPasswordOptions`.

WDYT?

Reply:
We can always move it later; resolving for now.